tests/integration/test_charm.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import pathlib
+
+import pytest
+import requests
+import tenacity
+import yaml
+from pytest_operator import plugin as pytest_plugin
+
+logger = logging.getLogger(__name__)
+
+LEGEND_HOST = "finos-legend"
+GITLAB_CLIENT_ID = "fake-client"
+GITLAB_CLIENT_SECRET = "fake-secret"
+
+METADATA = yaml.safe_load(pathlib.Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+
+MONGODB_APP_NAME = "mongodb-k8s"
+ENGINE_APP_NAME = "finos-legend-engine-k8s"
+SDLC_APP_NAME = "finos-legend-sdlc-k8s"
+STUDIO_APP_NAME = "finos-legend-studio-k8s"
+LEGEND_DB_APP_NAME = "finos-legend-db-k8s"
+GITLAB_INTEGRATOR_APP_NAME = "finos-legend-gitlab-integrator-k8s"
+NGINX_INGRESS_CHARM = "nginx-ingress-integrator"
+
+# TODO: Add STUDIO_APP_NAME to the list below once the following issue has been resolved:
+# https://github.com/finos/legend-studio/issues/1028
+LEGEND_APPS = [ENGINE_APP_NAME, SDLC_APP_NAME]
+LEGEND_APPS_CONFIG = [ENGINE_APP_NAME, SDLC_APP_NAME, STUDIO_APP_NAME]
+
+APP_LIST = [
+    MONGODB_APP_NAME,
+    LEGEND_DB_APP_NAME,
+    SDLC_APP_NAME,
+    ENGINE_APP_NAME,
+    STUDIO_APP_NAME,
+    GITLAB_INTEGRATOR_APP_NAME,
+]
+USEFUL_APPS = [
```

Comment:
All apps are useful. Considering that this list basically refers to the rest of the applications deployed alongside SDLC, i'd rename this to ``OTHER_APPS``.

Reply:
Solved.